.github/workflows/build-deploy-preview.yml

```
@@ -0,0 +1,146 @@
+        with:
+          issue-number: ${{ github.event.client_payload.pull_request.number }}
+          body: |
+            Tests running at: <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}>.
+            Workflow: `${{ github.workflow }}`.
+            PR: ${{ github.event.client_payload.pull_request.number }}.
+            URL: ${{ github.event.client_payload.pull_request.number }}ce.appsmith.com
+
+  server-build:
+    name: server-build
+    uses: ./.github/workflows/server-build.yml
+    secrets: inherit
+    with:
+      pr: ${{ github.event.client_payload.pull_request.number }}
+
+  client-build:
+    name: client-build
+    uses: ./.github/workflows/client-build.yml
+    secrets: inherit
+    with:
+      pr: ${{ github.event.client_payload.pull_request.number }}
+
+  rts-build:
+    name: rts-build
+    uses: ./.github/workflows/rts-build.yml
+    secrets: inherit
+    with:
+      pr: ${{ github.event.client_payload.pull_request.number }}
+
+  pushImage:
+    needs: [client-build, rts-build, server-build]
+    runs-on: ubuntu-latest
+
+    if: success()
+
+    steps:
+
+      - name: Set up Depot CLI
+        uses: depot/setup-action@v1
+
+      # Check out merge commit
+      - name: Checkout PR
+        uses: actions/checkout@v3
+        with:
+          ref: "refs/pull/${{ github.event.client_payload.pull_request.number }}/merge"
+
+      # Timestamp will be used to create cache key
+      - id: timestamp
+        run: echo "timestamp=$(date +'%Y-%m-%dT%H:%M:%S')" >> $GITHUB_OUTPUT
+
+      - name: Dump steps context
+        env:
+          STEPS_CONTEXT: ${{ toJson(steps) }}
+        run: echo "$STEPS_CONTEXT"
+
+      # get Git-hash will be used to create cache key
+      - id: git_hash
+        run: echo "git_hash=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Download the react build artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+          path: app/client/build
+
+      - name: Download the server build artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+          path: app/server/dist
+
+      - name: Download the rts build artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: rts-dist
+          path: app/rts/dist
+
+      - name: Untar the rts folder
+        run: |
+          tar -xvf app/rts/dist/rts-dist.tar -C app/rts/
+          echo "Cleaning up the tar files"
+          rm app/rts/dist/rts-dist.tar
+
+      - name: Push to Docker Hub
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+          repository: ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce
+          tags: ${{ steps.git_hash.outputs.git_hash }}
+    outputs:
+     imageHash: ${{ steps.git_hash.outputs.git_hash }}
+
+  buildDeployPreview:
```

Nit. We are following kebab case in CI YML files.
```suggestion
  build-deploy-preview:
```